integreat_compass/cms/views/index.py

```diff
@@ -1,9 +1,125 @@
+from ast import literal_eval
+
+from django.shortcuts import render
 from django.views.generic import TemplateView
 
+from ..constants import offer_group_types, offer_mode_types
+from ..models import Offer, Tag
+
 
 class IndexView(TemplateView):
     """
-    Start page of the project
+    View to see all offers
     """
 
     template_name = "index.html"
+
+    def get(self, request, *args, **kwargs):
+        """
+        This functions displays all offers
+
+        :param request: The current request
+        :type request: ~django.http.HttpRequest
+
+        :param args: The supplied arguments
+        :type args: list
+
+        :param kwargs: The supplied keyword arguments
+        :type kwargs: dict
+
+        :return: The rendered template response
+        :rtype: ~django.template.response.TemplateResponse
+        """
+        offers = Offer.objects.all()
+        tags = Tag.objects.all()
+        list_of_offers = []
+        for offer in offers:
+            list_of_offers.append(self.store_offer_data_in_dict(offer))
+        amount_of_offers = len(list_of_offers)
+
+        return render(
+            request,
+            self.template_name,
+            {
+                "tags": tags,
+                "offers": list_of_offers,
+                "amount_of_offers": amount_of_offers,
+                "offer_group_types": offer_group_types.CHOICES,
+                "offer_mode_types": offer_mode_types.CHOICES,
+            },
+        )
+
+    def post(self, request):
+        """
+        This functions filters
+
+        :param request: The current request
+        :type request: ~django.http.HttpRequest
+
+        :return: The rendered template response
+        :rtype: ~django.template.response.TemplateResponse
+        """
+        offers = Offer.objects.all()
+        tags = Tag.objects.all()
+
+        selected_filters = {
+            "tags": request.POST.getlist("offer-tags"),
+            "mode_type": request.POST.get("mode-type"),
+            "group_type": request.POST.get("group-type"),
+            "search": request.POST.get("offer-search"),
+        }
+
+        filtered_offers = []
+        selected_filters["tags"] = [literal_eval(i) for i in selected_filters["tags"]]
```

Comment:
Even if `literal_eval` is less scary than the evil `eval()`, it's not a function you should use often, since the input is completely user controlled and if you expect the input to have a certain type (in this case an `int`), you might encounter strange edge cases in your code, when the input suddenly is a completely different Python data structure.

Anyways, in this case the problem you are trying to solve is "I get string as input and want integers as output", so I don't really understand how you're ending up with `literal_eval()` as solution instead of just `int()`. Also, Django already provides a mechanism for this task: forms. If you want to have input forms and retrieve the input data as Python types other than strings, please create a form class, define the field as `IntegerField`, `ForeignKey`, etc. and access `form.cleaned_data` to get the user input as python data structure.

Apart from that, you don't even need to do this, since Django's queryset filters also work with string input, e.g. `Offer.objects.filter(tags="2")` is totally fine.

Reply:
Thank you for explaining, but this resolved itself since I switched to the Django filters :) I will keep it in mind for the future!